### DISCUSSION TITLE
Can't get window sort to work

### DISCUSSION THREAD

**Discussion by s-aleck** [davatorium/rofi#1467] [Q&A] [answered]
I'm running KDE Plasma on Manjaro.

From my `config.rasi`:

``` modi: "window,drun";
 theme: "solarized";
 font: "hack 10";
 sort: true;
 sorting-method: "normal";
 combi-modi: "window,run,drun,ssh";
 disable-history: true;
 window-format: "{w}    {c}   {t}";
 drun-match-fields: "name,generic,exec,categories,keywords";
 drun-display-format: "{name} [<span weight='light' size='small'><i>({generic})</i></span>]";
```

```  }
  filebrowser {
      directories-first: true;
      sorting-method: "name";
  }
```

I'm starting `rofi` with keyboard shortcuts with the following actions: `rofi -show window`, `rofi -show drun`.

Ideally, I would like the windows in the switcher to be sorted first by desktop, then by app, and then by file name. How can I achieve this behavior?

Maybe I would settle for grouping the windows by the app.

**Comment by DaveDavenport** [comment, marked as answer]
The default order is the order the window manager sets. There is no active sorting (before filtering) the list.

**Reply by s-aleck** [reply]
Thank you very much for your quick reply.

I'm looking into KWin's possibilities, but I must ask: is it possible to write a Python script for `rofi` which would sort the window list?

**Reply by DaveDavenport** [reply]
yeah, a custom modi should be no problem at all.
